rfcs/0089-relax-upstreaming.md

```diff
@@ -0,0 +1,775 @@
+
+## 4.5 PR list
+
+We plan to split the upstreaming into the following manageable PRs for TVM community review:
+
+- Relax IR
+- Relax VM
+- BlockBuilder
+- ExprFunctor/ExprVisitor/ExprMutator/IRFunctor
+- Relay → Relax translator
+- Minimum build (4 passes)
+- VM Codegen
+- E2E model compilation + execution
+
+# 5. **Future work**
+
+This RFC only focuses on the foundation part of Relax. After it, we will incrementally incorporate the additional capabilities and features. Relax aims to achieve parity with the functionality provided by Relay: this means that workloads which are functional on Relay will also be functional on Relax, even though the infrastructure underneath may change.
+
+Future plans that we will bring in future RFCs:
+
+- AOT: AOT compilation has a wide range of benefits such as being more space efficient and is necessary for resource-constrained projects like uTVM. We are committed to continuously supporting the AOT compilation in Relax, and there is an ongoing effort to connect Relax to the current AOT executor.
+- BYOC: We will try to reuse the existing translation spec. In Relax, BYOC can be formalized as a pass that call external packed functions.
+- Automatic scheduling: We will try to reuse the underlying [MetaSchedule](https://github.com/apache/tvm-rfcs/blob/main/rfcs/0005-meta-schedule-autotensorir.md) infrastructure, and the automatic scheduling can be formalized as a pass in Relax.
+
+We will migrate to use the new TVMScript parser/printer proposed in the following RFCs:
+
+- Parser: https://github.com/apache/tvm-rfcs/pull/79
+- Printer: https://github.com/apache/tvm-rfcs/pull/74
+
+# 6. **Rationale and alternatives**
+This proposal brings Relax as a separate dialect to the TVM ecosystem. We propose to do so because: Relax presents a set of closely chosen co-design choices to resolve pain points we observed so far in the TVM ecosystem, namely dynamic shape, side-effect handling, and cross-IR interaction. Additionally, the current Relax mainly focuses on graph-TIR-based cross-layer optimizations and complements the passes implemented in Relay.
+
+Presenting optional modules and dialects to the ecosystem is an important way to welcome new improvements, land our technical commitment timely, continue to reinvent ourselves, and welcome new community members who have new use cases.  It also aligns with practices we see in other ML projects, such as MLIR dialects, PyTorch FX modules, etc.
+
+As in these previous cases (MLIR dialects and TorchFX), there can be some amount of perceived duplications with existing modules. Nevertheless, there are usually technical reasons for a new module, both from design consistency, time to delivery, and separation of concerns from stable modules during improvements. Additionally, there is also a majority amount of support to help continue maintaining the new module. This new community effort would help us to welcome new members to the community, and enhance the overall development of TVM.
```

Comment:
In previous comments I was unfamiliar with the project guidelines for MLIR and Torch, but reading their contribution guidelines, they actually require much more certainty on the future landscape for a given feature to make into the codebase, so I see no reason why we should have lower standards then those communities, at the same time we mention those communities as examples to be followed.

Based on that, I disagree on the premise that features should be brought into the codebase on the major justification of welcoming new contributors.